ReactWindows/ReactNative/Views/Image/ReactImageManager.cs

```diff
@@ -244,5 +264,94 @@ private void OnImageOpened(Border view)
                     view.GetTag(),
                     ReactImageLoadEvent.OnLoadEnd));
         }
+
+        /// <summary>
+        /// Set the source URI of the image.
+        /// </summary>
+        /// <param name="view">The image view instance.</param>
+        /// <param name="source">The source URI.</param>
+        private void SetUriFromSingleSource(Border view, string source)
+        {
+            var imageBrush = (ImageBrush)view.Background;
+            var tag = view.GetTag();
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(
+                    new ReactImageLoadEvent(
+                        tag,
+                        ReactImageLoadEvent.OnLoadStart));
+
+            var reference = _imageCache.Get(source);
+            var subscription = reference.LoadedObservable.Subscribe(
+                _ =>
+                {
+                    imageBrush.ImageSource = reference.Image;
+                    OnImageOpened(view);
+                },
+                _ => OnImageFailed(view));
+
+            var disposable = default(SerialDisposable);
+            if (!_disposables.TryGetValue(tag, out disposable))
+            {
+                disposable = new SerialDisposable();
+                _disposables.Add(tag, disposable);
+            }
+
+            disposable.Disposable = new CompositeDisposable
+            {
+                reference,
+                subscription,
+            };
+        }
+
+        /// <summary>
+        /// Chooses the uri with the size closest to the target image size. Must be called only after the
+        /// layout pass when the sizes of the target image have been computed, and when there are at least
+        /// two sources to choose from.
+        /// </summary>
+        /// <param name="view">The image view instance.</param>
+        private void SetUriFromMultipleSources(Border view)
+        {
+            var sources = default(List<KeyValuePair<string, double>>);
+            if (_imageSources.TryGetValue(view.GetTag(), out sources))
+            {
+                var targetImageSize = view.Width * view.Height;
+                var left = 0;
+                var right = sources.Count - 1;
+                var bestResult = default(KeyValuePair<string, double>);
+                do
+                {
+                    var mid = (left + right) >> 1;
+                    bestResult = sources[mid];
+                    if (sources[mid].Value < targetImageSize)
```

Comment:
>sources[mid] [](start = 24, length = 12)

Use `bestResult` variable rather than another array lookup.  Any chance we could put this in a helper method and unit test?